frontend/tests/jest/tests/shopify-storefront-sdk.test.tsx

```diff
@@ -0,0 +1,66 @@
+import { SentryError } from '@ifixit/sentry';
+import {
+   getServerShopifyStorefrontSdk,
+   ShopCredentials,
+} from '../../../lib/shopify-storefront-sdk/index';
+import { Response } from 'node-fetch';
+
+jest.mock('node-fetch');
+
+describe('getServerShopifyStorefrontSdk', () => {
+   it('throws a SentryError when fetch result has errors', async () => {
+      const mockFetch = jest.fn();
+      (global as any).fetch = mockFetch;
+
+      const mockResponse: Partial<Response> = {
+         ok: false,
+         statusText: 'Not Found',
+         json: async () => ({
+            errors: [
+               {
+                  message: 'Error message',
+                  extensions: {
+                     code: 'ERROR_CODE',
+                  },
+               },
+            ],
+         }),
+      };
+
+      mockFetch.mockResolvedValue(mockResponse);
+
+      const shop: ShopCredentials = {
+         shopDomain: 'test.myshopify.com',
+         storefrontDelegateToken: 'test_token',
+      };
+
+      try {
+         await getServerShopifyStorefrontSdk(shop);
+      } catch (error) {
+         expect(error).toBeInstanceOf(SentryError);
```

Comment:
I'm pretty sure this wasn't a connection error, nor a 500, but an in-band error on a 200.

Reply:
Thanks for the clarity. We end up throwing if it was `response.ok` and also `response.json().errors`. I have adjusted the test to show we would throw and the error would contain the correct data

Reply:
Right now, it is double logged due to this log: https://github.com/iFixit/react-commerce/blob/e4cfd30d4d20baaf6c2b0db25a75c598ad834496/frontend/lib/shopify-storefront-sdk/index.ts#L44

It would only be sent to Sentry once.

It's tricky to know what is correct. The withCache is quite generic trying to ensure it logs errors. The passed-in functions to get data don't have any guarantee they log errors.

I am inclined to leave it. The benefit is calling the fetch without caching will log.

Otherwise, I'd assume the withCache will log for you.